### PR TITLE
Implement support for multiple UARTS on i.MX6

### PR DESCRIPTION
**Pull request by Deryugin**
Configure pins on iwave rainbow g15s to use UARTs on 84-pin extension.

### CODE REVIEW

File: src/drivers/serial/imx_uart/imx_uart.c [hunk 1]
```diff
@@ -69,49 +153,49 @@ static int imxuart_getc(struct uart *dev) {
 }
 
 static int imxuart_putc(struct uart *dev, int ch) {
-	while (!(UART(USR2) & USR2_TXFE)) {
-	}
-
 	UART(TXR) = ch;
 
+	while (!(UART(UTS) & (1 << 6))) {
```
Comment (anton-bondarev) [inline src/drivers/serial/imx_uart/imx_uart.c]: Magic constant?
Why are you wait free tx buffer?

File: src/drivers/serial/imx_uart/imx_uart.c [hunk 2]
```diff
@@ -11,40 +11,124 @@
...
+	/* Reset */
+	UART(UCR2) = 0;
+	while(UART(UTS) & UTS_RST);
+
+	UART(UCR1) = UCR1_UARTEN;
+	UART(UCR2) = UCR2_NORESET | UCR2_RXEN | UCR2_TXEN | UCR2_WORDSIZE | UCR2_IRTS;
+	UART(UCR3) = UCR3_RXDMUXSEL | UCR3_RI | UCR3_DCD | UCR3_DSR;
+	UART(UCR4) = 0;
+	UART(UFCR) = 4 << UFCR_RFDIF;
+	UART(UBIR) = 0x0F;
```
Comment (anton-bondarev) [inline src/drivers/serial/imx_uart/imx_uart.c]: Is it for a frequency. May be formula for calculating this fields?

File: src/drivers/serial/imx_uart/imx_uart.c (hunk 1 above)
Comment (Deryugin) [inline src/drivers/serial/imx_uart/imx_uart.c]: Fixed constant in the following commit.
Waiting is just to make sure symbol was sent, nothing special.

File: src/drivers/serial/imx_uart/imx_uart.c (hunk 2 above)
Comment (Deryugin) [inline src/drivers/serial/imx_uart/imx_uart.c]: This frequency is tied to reference clock, and I don't know how to configure it. So I suggest just to leave it as is.

**Pull request merged**